src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -108,9 +120,17 @@ public HostContext(HostType hostType, string logFile = null)
             _loadContext = AssemblyLoadContext.GetLoadContext(typeof(HostContext).GetTypeInfo().Assembly);
             _loadContext.Unloading += LoadContext_Unloading;
 
-            this.SecretMasker.AddValueEncoder(ValueEncoders.JsonStringEscape, $"HostContext_{WellKnownSecretAliases.JsonStringEscape}");
+            // Having created one of three secret maskers, we now need to supplement the core masking capability with
+            // additional checks. The original VSO secret masker has no additional redactions specific to credentials.
+
+            if (!useOssSecretMasker)
+            {
+                this.AddAdditionalMaskingRegexes();
```

Comment:
If I understood correctly, this method is no longer meant to be called from outside this class. Consider making it private. Also consider rename to AddCredScanPatterns since that's all it adds.

Reply:
Hmm, it's an extension method and visible outside the assembly, maybe someone else is calling it?

Reply:
But if so, this will add duplicate patterns to OSS patterns. Maybe the if needs to move inside here.